File: python/graph/graph_test.py
from .graph import Graph


def test_graph():
    n = 5
    directed = False
    g = Graph(n, directed)

    assert g.get_vertices_num() == n
    assert g.is_directed() == directed

    g.add_edge(0, 1)
    g.add_edge(1, 2)
    g.add_edge(1, 3)
    g.add_edge(1, 4)
    g.add_edge(2, 3)
    g.add_edge(4, 2)

    assert g.is_edge(0, 1)
    assert not g.is_edge(0, 2)
    assert g.get_edge_weight(0, 1) == 1

    g.remove_edge(0, 1)
    assert not g.is_edge(0, 1)

File: go/graph/graph_test.go
package graph

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestGraph(t *testing.T) {
	var n = 5
	var directed = false
	g := NewGraph(n, directed)

	assert.Equal(t, n, g.n)
	assert.Equal(t, directed, g.directed)

	g.addEdge(0, 1, 1)
	g.addEdge(1, 2, 1)
	g.addEdge(1, 3, 1)
	g.addEdge(1, 4, 1)
	g.addEdge(2, 3, 1)
	g.addEdge(4, 2, 1)

	assert.True(t, g.isEdge(0, 1))
	assert.False(t, g.isEdge(0, 2))
	assert.Equal(t, 1, g.getEdgeWeight(0, 1))

	g.removeEdge(0, 1)
	assert.False(t, g.isEdge(0, 1))
}
